app/assets/javascripts/calendar/calendar.js

```diff
@@ -0,0 +1,61 @@
+$(function () {
+    console.log(events);
+
+   events.forEach(function (block) {
+        block.forEach(function (event) {
+            var tableCell = getEventTableCell(event);
+            var eventContainer = tableCell.children('.event-container');
+            setEventSize(eventContainer, block, event);
+
+            var length = eventLengthInSeconds(event);
+
+            $.ajax({
+                type: 'GET',
+                dataType: 'html',
+                url: eventUrl + '/' + event.id
+            }).done(function (data) {
+                eventContainer.append(data);
+                var height = length * parseInt(tableCell.css('height'));
+                getEventElement(eventContainer, event).css({height: height + 'px'});
+            });
+        });
+    });
+
+    $(window).resize(function () {
+        events.forEach(function (block) {
+            block.forEach(function (event) {
+                var tableCell = getEventTableCell(event);
+                var eventContainer = tableCell.children('.event-container');
+                setEventSize(eventContainer, block, event);
+            });
+        });
+    });
+
+    function eventWidth(block) {
+        return parseInt(getEventTableCell(block[0]).css('width')) / block.length;
+    }
+
+    function getEventTableCell(event) {
+        return $("td[data-date='" + new Date(event.start_time).getTime() / 1000 + "']");
+    }
+
+    function getEventElement(eventContainer, event) {
+        return eventContainer.children('.event[data-id="' + event.id + '"]');
+    }
+
+    function setEventSize(eventContainer, block, event) {
+        var width = eventWidth(block);
```

Comment:
Currently we can calculate this just once for the block, since it's the same for every event in particular block.